ext/xml/xml.c

```diff
@@ -1266,21 +1323,21 @@ PHP_FUNCTION(xml_parse_into_struct)
 	}
 
 	if (info) {
-		info = zend_try_array_init(info);
-		if (!info) {
+		if (!zend_try_array_init(info)) {
 			RETURN_THROWS();
 		}
 	}
 
-	xdata = zend_try_array_init(xdata);
-	if (!xdata) {
+	if (!zend_try_array_init(xdata)) {
```

Comment:
not entirely certain about these two changes, is it a cleanup thing you re doing here ?

Reply:
It's not cleanup, we need the original value of xdata.
xdata is now a reference, but if the return value of `zend_try_array_init` were used then it would be the array that xdata references _to_. We need to hold on to the reference, we can't hold on to the array because that would break the RC1 constraint of the array and would also make it impossible to separate it.

Reply:
ah ok makes a lot more sense.